mqtt/mqtt-broker-core/src/auth/authentication.rs

```diff
@@ -82,44 +128,50 @@ pub struct DefaultAuthenticator;
 impl Authenticator for DefaultAuthenticator {
     type Error = Box<dyn StdError>;
 
-    async fn authenticate(
-        &self,
-        _: Option<String>,
-        _: Credentials,
-    ) -> Result<Option<AuthId>, Self::Error> {
+    async fn authenticate(&self, _: AuthenticationContext) -> Result<Option<AuthId>, Self::Error> {
         Ok(None)
     }
 }
 
 #[cfg(test)]
 mod tests {
+    use std::net::SocketAddr;
+
     use matches::assert_matches;
 
-    use crate::auth::{
-        authenticate_fn_ok, AuthId, Authenticator, Credentials, DefaultAuthenticator,
+    use super::{
```

Comment:
I thought we were not in favor of `super` :)

Reply:
Yeah, I am moving code across modules for the last few weeks, and changed my opinion:
1. when in unit tests we use **parent** module items: struct, enums, functions - use `super`
2. whatever outside of parent module, use `crate::module::Struct`